src/soroban/contract_client.ts

```diff
@@ -42,14 +37,61 @@ export type ContractClientOptions = {
    * })
    * ```
    */
-  wallet: Wallet;
+  wallet?: Wallet;
+
+  /**
+   * You can pass in `wallet` OR `account`, but not both. If you only pass
+   * `wallet`, `account` will be derived from it. If you can bypass this
+   * behavior by passing in your own account object.
+   */
+  account?: Account | Promise<Account>;
+};
+
+async function getPublicKey(wallet?: Wallet): Promise<string | undefined> {
+  if (!wallet) return undefined;
+  if (!(await wallet.isConnected()) || !(await wallet.isAllowed())) {
+    return undefined;
+  }
+  return (await wallet.getUserInfo()).publicKey;
+};
+
+export const NULL_ACCOUNT =
+  "GAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWHF";
+
+/**
+ * Get account details from the Soroban network for the publicKey currently
+ * selected in user's wallet. If user is not connected to their wallet, {@link
+ * getPublicKey} returns undefined, and this will return {@link NULL_ACCOUNT}.
+ * This works for simulations, which is all that's needed for most view calls.
+ * If you want the transaction to be included in the ledger, you will need to
+ * provide a connected wallet.
+ */
+export async function getAccount(server: SorobanRpc.Server, wallet?: Wallet): Promise<Account> {
```

Comment:
should/could this be the `AcceptsWalletOrAccount` interface, instead?

Reply:
I'm not sure I understand the question. This function is used during initialization of the ContractClient and during initialization of an AssembledTransaction to get the account from the passed-in Wallet (or Signer, or whatever we want to call it), if you don't pass in an explicit `Account`. So it is used to support the `AcceptsWalletOrAccount` interface.

Reply:
I got rid of `AcceptsWalletOrAccount` altogether now. See updated PR description.